core/templates/pages/contributor-dashboard-page/translation-opportunities/translation-opportunities.component.spec.ts

```diff
@@ -16,57 +16,83 @@
-
-describe('Translation opportunities component', function() {
-  var ctrl = null;
-  var $q = null;
-  var $rootScope = null;
-  var $scope = null;
-  var $uibModal = null;
-  var contributionOpportunitiesService = null;
-  var siteAnalyticsService = null;
-  var translationLanguageService = null;
-  var userService = null;
-
-  var opportunitiesArray = [];
+import { ComponentFixture, fakeAsync, TestBed, tick } from '@angular/core/testing';
+import { HttpClientTestingModule, HttpTestingController } from '@angular/common/http/testing';
+import { NgbActiveModal, NgbModal, NgbModalRef } from '@ng-bootstrap/ng-bootstrap';
 
-  importAllAngularServices();
-
-  beforeEach(function() {
+import { ContributionOpportunitiesService } from 'pages/contributor-dashboard-page/services/contribution-opportunities.service';
+import { ExplorationOpportunitySummary } from 'domain/opportunity/exploration-opportunity-summary.model';
+import { OpportunitiesListComponent } from 'pages/contributor-dashboard-page/opportunities-list/opportunities-list.component';
+import { OpportunitiesListItemComponent } from 'pages/contributor-dashboard-page/opportunities-list-item/opportunities-list-item.component';
+import { TranslationLanguageService } from 'pages/exploration-editor-page/translation-tab/services/translation-language.service';
+import { TranslationModalComponent } from 'pages/contributor-dashboard-page/modal-templates/translation-modal.component';
+import { TranslationOpportunitiesComponent } from './translation-opportunities.component';
+import { UserInfo } from 'domain/user/user-info.model';
+import { UserService } from 'services/user.service';
+import { WrapTextWithEllipsisPipe } from 'filters/string-utility-filters/wrap-text-with-ellipsis.pipe';
+import { LazyLoadingComponent } from 'components/common-layout-directives/common-elements/lazy-loading.component';
+import { SchemaBasedEditorDirective } from 'components/forms/schema-based-editors/schema-based-editor.directive';
+import { AngularHtmlBindWrapperDirective } from 'components/angular-html-bind/angular-html-bind-wrapper.directive';
+import { CkEditorCopyToolbarComponent } from 'components/ck-editor-helpers/ck-editor-copy-toolbar/ck-editor-copy-toolbar.component';
+
+describe('Translation opportunities component', () => {
+  let contributionOpportunitiesService: ContributionOpportunitiesService;
+  let translationLanguageService: TranslationLanguageService;
+  let userService: UserService;
+  let modalService: NgbModal;
+  let component: TranslationOpportunitiesComponent;
+  let fixture: ComponentFixture<TranslationOpportunitiesComponent>;
+  let translationModal: NgbModalRef;
+  let httpTestingController;
+  let loggedInUserInfo = new UserInfo(
+    false, false, false, false, false,
+    'en', 'username', 'test@example.com', true
+  );
+  const notLoggedInUserInfo = new UserInfo(
+    false, false, false, false, false,
+    'en', null, null, false
+  );
+
+  let opportunitiesArray = [];
```

Comment:
Can we use `const` here?

Reply:
No, the array is assigned a value on line 98 in the same file.